browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -31,33 +31,26 @@ void ApplyPotentialQueryStringFilter(std::shared_ptr<BraveRequestInfo> ctx) {
     return;
   }
 
-  if (ctx->method != "GET") {
-    return;
-  }
-
+  // We either pass a valid redirect_source or initiator_url
   if (ctx->redirect_source.is_valid()) {
     if (ctx->internal_redirect) {
       // Ignore internal redirects since we trigger them.
       return;
     }
 
-    if (net::registry_controlled_domains::SameDomainOrHost(
-            ctx->redirect_source, ctx->request_url,
-            net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES)) {
-      // Same-site redirects are exempted.
-      return;
+    auto filtered_url = ApplyPotentialQueryStringFilter(
```

Comment:
overloading this method name is confusing here, but fixed by https://github.com/brave/brave-core/pull/19633/files#r1289153573 which is the standard naming convention.

Reply:
also `OnBeforeURLRequest_SiteHacksWork` should just call `query_filter::MaybeApplyQueryStringFilter` (allow_brave_shields check can be added after `ctx->request_url.has_query()` to get rid of this method

Reply:
I haven't done this yet but I can revisit.

Reply:
Ok did this